src/main/java/org/sudoku/Cell.java

```diff
@@ -1,99 +1,63 @@
 package org.sudoku;
 
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
-
-public class Cell implements Comparable {
-
-  private Set<Lane> belongToLanes = new HashSet<>(3);
-  private Set<Byte> possibleNumbers = new HashSet<>(9);
-  private Byte value = null;
-  private byte positionX;
-  private byte positionY;
-
-  public static final Byte[] numbers = {1,2,3,4,5,6,7,8,9};
-
-  public Cell(byte positionX, byte positionY) {
-    this.positionX = positionX;
-    this.positionY = positionY;
-    this.possibleNumbers.addAll(Arrays.asList(numbers));
-  }
-
-
-  public void setBelongToLanes(Lane vLane, Lane hLane, Lane square) {
-    this.belongToLanes.add(vLane);
-    this.belongToLanes.add(hLane);
-    this.belongToLanes.add(square);
-  }
-
-  public Set<Byte> getPossibleNumbers() {
-    return possibleNumbers;
-  }
-
-  public Set<Cell> setNumberValue(byte value) {
-    if(this.value!=null) {
-      throw new SudokuException("Tried to set \""+value+"\" for "+this.debug());
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cell {
+    private int y;
+    private int x;
+    private int value;
+    public List<Integer> possibleValues;
```

Comment:
why this is List and public?
broken incapsulation and inefficient data structure